engine/src/execution_graph/logic_controllers/taskflow/kernel.h

```diff
@@ -125,6 +127,10 @@ class kernel {
 	std::shared_ptr<graph> query_graph;
 	std::shared_ptr<Context> context;
 
+	// useful when the Algebra Relacional only contains: LogicalTableScan and LogicalLimit
+	std::atomic<bool> has_limit_;
```

Comment:
why are these atomic? these are only set once in a single threaded part of the code and then they are never modified, right?

Reply:
Yep, you are right. It's only set once and never modified. I just remove the `std::atomic` part . 
Also I tested this feature on the dgx1.